nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -236,6 +254,38 @@ private NullnessStore methodInitialStore(
     return result.build();
   }
 
+  /**
+   * @param classTree a class
+   * @return the nullness info of locals in the enclosing environment for the closest enclosing
+   *     local or anonymous class. if no such class, returns an empty {@link NullnessStore}
+   */
+  private NullnessStore getEnvNullnessStoreForClass(ClassTree classTree) {
+    NullnessStore envStore = NullnessStore.empty();
+    ClassTree enclosingLocalOrAnonymous = findEnclosingLocalOrAnonymousClass(classTree);
+    if (enclosingLocalOrAnonymous != null) {
+      EnclosingEnvironmentNullness environmentNullness =
+          EnclosingEnvironmentNullness.instance(context);
+      envStore =
+          Objects.requireNonNull(
+              environmentNullness.getEnvironmentMapping(enclosingLocalOrAnonymous));
+    }
+    return envStore;
+  }
+
+  @Nullable
+  private ClassTree findEnclosingLocalOrAnonymousClass(ClassTree classTree) {
+    Symbol.ClassSymbol symbol = ASTHelpers.getSymbol(classTree);
+    while (symbol.getNestingKind().isNested()) {
```

Comment:
Curious why this is a `while` loop. Can the result of `symbol.owner.enclClass()` be anything other than the closest enclosing class? If so... comment?

Reply:
Added a comment on why we need the loop, and the test input `NestedClassInAnonymous` inside `CapturingScopes`